### PR TITLE
Measure on gpu

### PR DESCRIPTION
**Pull request by tfarago**
Closely related to #155, this shifts the statistics to GPU. It's very preliminary for now and in the end, only one of these two PRs should survive.

### CODE REVIEW

**Comment by tfarago** [comment]
The vertical reduction is very inefficient, but I am confident that want to leave some room for improvement :wink:

**Comment by tfarago** [comment]
I wanted to write some tests, but with the signal as result I am clueless. @matze?

File: src/ufo-measure-task.c [hunk 1]
```diff
@@ -28,16 +35,49 @@
...
-    M_STD,
     M_MIN,
     M_MAX,
-    M_LAST
+    M_SUM,
+    M_MEAN,
+    M_VAR,
+    M_STD,
+    M_SKEW,
+    M_KURTOSIS,
```
Comment (matze) [inline src/ufo-measure-task.c]: Could you still use `M_LAST` here and in the static data structures? It's easier to understand and in the long run easier to extend.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -205,12 +205,6 @@ if (UCA_INCLUDE_DIRS AND UCA_LIBRARIES)
     link_directories(${UCA_LIBRARY_DIRS})
 endif ()
 
-if (GSL_FOUND)
-    include_directories(${GSL_INCLUDE_DIRS})
-    list(APPEND measure_aux_LIBS ${GSL_LIBRARIES})
-    list(APPEND ufofilter_SRCS ufo-measure-task.c)
-endif ()
```
Comment (matze) [inline src/CMakeLists.txt]: Could you also remove the `gsl_dep` check in `meson.build`?

Comment (tfarago) [reply src/CMakeLists.txt]: Done in 1fa1e295d1a41804ba4ca55daa7a92866162d092

**Comment by matze** [comment]
Is this done? If so we could merge it …

**Comment by tfarago** [comment]
Once all the tests pass...

**Comment by tfarago** [comment]
Tests pass, I can squash once reviewed.

**Comment by matze** [comment]
Do you think it's worth keeping the debug output? Other than that it looks as good as three weeks ago ;-)

**Comment by tfarago** [comment]
Squashed and good to go. I would keep the output, one can turn it off, so it shouldn't be too annoying.

**Comment by tfarago** [comment]
Oh wait, one more thing. If I have more GPUs in the system enabled the callback doesn't get called, which will be a serious problem in real life. So we either have to fix it or use normal `UfoBuffer` for output.

**Comment by tfarago** [comment]
Any thoughts on the previous comment @matze ?

**Comment by matze** [comment]
Ah yeah, let me see why the callback is not delivered.

**Comment by tfarago** [comment]
@matze can you check the last commit? I will change the tests accordingly and then we could merge.

**Comment by tfarago** [comment]
Aaaand it also works without limiting the number of GPUs to 1, so it's really good to go. Tests are done as well and passed.

**Comment by matze** [comment]
Looks nice and tidy :+1:

**Pull request merged**